docs/DeveloperGuide.md

```diff
@@ -1,34 +1,429 @@
+Step 3. The UI then prompts the user to input the next block that is in the day's schedule.  <br /> The inputted location will be appended to an ArrayList. <br />
+Step 4. Repeat step 3 until the word `END` is input by the user. <br /> 
+Step 5. The inputted day, and the filled Arraylist from step 3 is then passed into the DailyRoute object<br /> This done using the addDailyRoute method from the DailyRoute class <br />
+Step 6. The day and filled Arraylist passed in step 5 is then saved in a hashmap that the DailyRoute object contains. <br /> 
+
+The following image shows the sequence diagram in which the add day command is implemented
+![img.png](images/addday.png)
+
+Given below is an example usage scenario and how the showDailyRoute mechanism behaves at each step.
+
+Step 1. The user launches the application.<br />
+Step 2. The user executes `show daily route` command. UI will then prompt the user `SELECT ENTRY TO VIEW:`  to input a day index. This returns an arraylist of the day's schedule <br />
+Step 3. The routing algorithm is now performed for each of the blocks in the array list in order. Each execution of the routing algorithm returns a string which is then appended to the previous one. <br />
+Step 4. The String is output through Daily Route Ui <br />
+
+The following image shows the sequence diagram in which the add day command is implemented
+![img.png](images/showday.png)
+
+### 3.3. Finding The Shortest Route Feature
+#### Current Implementation
+
+The current implementation of finding the shortest route is facilitated by the `Router` class which uses data stored in `NusMap`, `Block`, and `BlockAlias` class to return the shortest path.
+
+The `GoCommand` class extends the `Command` class and overrides the `execute` method to run the routing algorithm.
+
+The image below depicts how the `GoCommand` is implemented.
+
+![img.png](images/router.png)
+
+Given below is an example scenario of how the routing algorithm functions.
+
+Step 1. User executes `GoCommand` and the `RouterUi` reads in the starting location and destination.<br />
+Step 2. `GoCommand` will then check if the second entry is eatery. If it is not "EATERY", step 3 and 4 are skipped for step 5.<br />
+Step 3. `GoCommand` will then create an instance of `EateryList` and invokes its method `sortEateriesByDistance()` which returns a list of eateries in order of the closest distance.<br />
+Step 4. `GoCommand` then takes in the selection of eatery that the user is chosen and sets the destination.<br/>
+Step 5. The Router will then run the `findShortestRoute()` method which is a routing algorithm based on breath-first search. This returns the shortest route as a string<br />
+Step 6. The `RouterUi` will then show the shortest route to the user through `showMessageWithDivider()` method.<br />
+
+Shown below is the sequence diagram when a valid block is entered for the starting location and destination.
+![img.png](images/routersequencediagram.png)
```

Comment:
![image](https://user-images.githubusercontent.com/57165946/113232031-f9fda500-92ce-11eb-92c9-3087dab7f3f6.png)
Is the block object kept? If not consider putting a cross at the bottom of the line